src/frontend/metrics/generateMetricsReport.ts

```diff
@@ -22,17 +23,25 @@ export default function generateMetricsReport({
       addToSet(countries, positionToCountries(lat, lon));
     }
   }
-
   return {
     type: 'metrics-v1',
     appVersion: packageJson.version,
     os,
     osVersion,
     screen: {width: screen.width, height: screen.height},
     ...(countries.size ? {countries: Array.from(countries)} : {}),
+    percentageOfNetworkAvailability:
+      getPercentageOfNetworkAvailability(observations),
   };
 }
 
-function addToSet<T>(set: Set<T>, toAdd: Iterable<T>): void {
-  for (const item of toAdd) set.add(item);
+export function getPercentageOfNetworkAvailability(
```

Comment:
Nitpick: could you mention that this is only exported for tests? Alternatively, moving it to its own file with tests would be great.

Reply:
I don't understand this. You mean, having a test file only for this function? Or, two files? (one with the implementation and a test file)